addons/binding/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/ThirdGeneration/ThirdGenerationHandler.java

```diff
@@ -0,0 +1,767 @@
+                 */
+                logger.info("Session expired");
+                try {
+                    authenticate();
+                } catch (Exception e) {
+                }
+                /*
+                 * Retry
+                 */
+                updateMessageContentResponse = executeHttpPost(PROCESSDATA, updateMessageJsonArray, sessionId);
+            }
+        } catch (Exception e) {
+            /*
+             * Communication problem
+             */
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, COMMUNICATION_ERROR_HTTP);
+            return;
+        }
+        JsonArray updateMessageResultsJsonArray;
+        try {
+            updateMessageResultsJsonArray = getJsonArrayFromResponse(updateMessageContentResponse);
+        } catch (Exception e) {
+            /*
+             * Communication problem
+             */
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, COMMUNICATION_ERROR_HTTP);
+            return;
+        }
+        /*
+         * Map the returned values back to the channels and update them
+         */
+        for (int i = 0; i < updateMessageResultsJsonArray.size(); i++) {
+            JsonObject moduleAnswer = updateMessageResultsJsonArray.get(i).getAsJsonObject();
+            String moduleName = moduleAnswer.get("moduleid").getAsString();
+            JsonArray processdata = moduleAnswer.get("processdata").getAsJsonArray();
+
+            for (int j = 0; j < processdata.size(); j++) {
+                /*
+                 * Update the channels with their new value
+                 */
+                JsonObject newValueObject = processdata.get(j).getAsJsonObject();
+                String valueId = newValueObject.get("id").getAsString();
+                double valueAsDouble = newValueObject.get("value").getAsDouble();
+                ThirdGenerationChannelMappingToWebApi channel = channelList.get(moduleName).stream()
+                        .filter(c -> c.moduleId.equals(moduleName) && c.processdataId.equals(valueId)).findFirst()
+                        .get();
+                updateChannelValue(channel.channelUID, channel.dataType, valueAsDouble);
+            }
+        }
+    }
+
+    /**
+     * Update the channel to the given value.
+     * The value is set to the matching data (SITypes etc)
+     *
+     * @param channeluid Channel to update
+     * @param dataType   target data type
+     * @param value      value
+     */
+    private void updateChannelValue(String channeluid, ThirdGenerationChannelDatatypes dataType, Double value) {
+        switch (dataType) {
+            case Integer: {
+                updateState(channeluid, new DecimalType(value.longValue()));
+                break;
+            }
+            case Percent: {
+                updateState(channeluid,
+                        new QuantityType<javax.measure.quantity.Dimensionless>(value, SmartHomeUnits.PERCENT));
```

Comment:
Don't use full paths, but add to imports:
```suggestion
                        new QuantityType<Dimensionless>(value, SmartHomeUnits.PERCENT));
```

Reply:
Done in next commit